prov/efa/src/efa_strerror.c

```diff
@@ -63,6 +69,24 @@ static const char *efa_io_comp_status_str(enum efa_errno status)
 		: status_str[status];
 }
 
+static inline char *efa_host_id_str(uint64_t host_id)
+{
+	if (!host_id) {
+		return NULL;
+	}
+
+	char *host_id_str = calloc(HOST_ID_LENGTH + 1, sizeof(host_id_str));
+	if (!host_id_str) {
+		return NULL;
+	}
```

Comment:
If caller should free this memory, it should be clearly stated in the doxygen block for this function.

I would prefer if this function wrote to a global variable (like fi_tostr) or to a caller-provided string, rather than allocate new memory.